src/ArkaneWalletSubProvider.ts

```diff
@@ -71,9 +71,10 @@ export class ArkaneWalletSubProvider extends BaseWalletSubprovider {
    * @return An array of accounts
    */
   public async getAccountsAsync(): Promise<string[]> {
-    let that = this;
     let promise: Promise<any>;
-    if (!this.authenticated) {
+
+    const authResult: AuthenticationResult = await this.arkaneConnect.checkAuthenticated();
```

Comment:
will this query keycloak each time or not?

Reply:
![image](https://user-images.githubusercontent.com/11631532/120348817-d5896d80-c2fd-11eb-9b3b-170ab3274930.png)
No, only if `loginResult` is not there yet (so basically the first time)